### ISSUE TITLE
2019-11-20 tc018 M4SS execution test case is displaying errors concerning specific gravity dose adjustment

### ISSUE THREAD

**Issue by tensfeldt** [tensfeldt/openNCA#33]
This code was not entirely complete in master branch.

Parameter results are generated but not entirely confirmed as yet.

errors displayed are:

```
> results_list <- run_computation(data=d, map=mct, flag=flags, parameterset=parameterset)
specific_gravity_adjustment : units not recognized!i:  1  tmp_dose:  30  d:  1  dose[[d]]:  30 
i:  1  tmp_dose:  30  d:  1  dose[[d]]:  30 
specific_gravity_adjustment : units not recognized!i:  2  tmp_dose:  200  d:  1  dose[[d]]:  200 
i:  2  tmp_dose:  200  d:  1  dose[[d]]:  200 
specific_gravity_adjustment : units not recognized!i:  3  tmp_dose:  200  d:  1  dose[[d]]:  200 
i:  3  tmp_dose:  200  d:  1  dose[[d]]:  200 
specific_gravity_adjustment : units not recognized!i:  4  tmp_dose:  200  d:  1  dose[[d]]:  200 
i:  4  tmp_dose:  200  d:  1  dose[[d]]:  200 
specific_gravity_adjustment : units not recognized!i:  5  tmp_dose:  200  d:  1  dose[[d]]:  200 
i:  5  tmp_dose:  200  d:  1  dose[[d]]:  200 
specific_gravity_adjustment : units not recognized!i:  6  tmp_dose:  100  d:  1  dose[[d]]:  100 
i:  6  tmp_dose:  100  d:  1  dose[[d]]:  100 
specific_gravity_adjustment : units not recognized!i:  7  tmp_dose:  30  d:  1  dose[[d]]:  30 
i:  7  tmp_dose:  30  d:  1  dose[[d]]:  30 
specific_gravity_adjustment : units not recognized!i:  8  tmp_dose:  200  d:  1  dose[[d]]:  200 
i:  8  tmp_dose:  200  d:  1  dose[[d]]:  200 
specific_gravity_adjustment : units not recognized!i:  9  tmp_dose:  30  d:  1  dose[[d]]:  30 
i:  9  tmp_dose:  30  d:  1  dose[[d]]:  30 
specific_gravity_adjustment : units not recognized!i:  10  tmp_dose:  200  d:  1  dose[[d]]:  200 
i:  10  tmp_dose:  200  d:  1  dose[[d]]:  200 
specific_gravity_adjustment : units not recognized!i:  11  tmp_dose:  100  d:  1  dose[[d]]:  100 
i:  11  tmp_dose:  100  d:  1  dose[[d]]:  100 
specific_gravity_adjustment : units not recognized!i:  12  tmp_dose:  200  d:  1  dose[[d]]:  200 
i:  12  tmp_dose:  200  d:  1  dose[[d]]:  200 
specific_gravity_adjustment : units not recognized!i:  13  tmp_dose:  200  d:  1  dose[[d]]:  200 
i:  13  tmp_dose:  200  d:  1  dose[[d]]:  200 
specific_gravity_adjustment : units not recognized!i:  14  tmp_dose:  30  d:  1  dose[[d]]:  30 
i:  14  tmp_dose:  30  d:  1  dose[[d]]:  30 
specific_gravity_adjustment : units not recognized!i:  15  tmp_dose:  100  d:  1  dose[[d]]:  100 
i:  15  tmp_dose:  100  d:  1  dose[[d]]:  100 
specific_gravity_adjustment : units not recognized!i:  16  tmp_dose:  30  d:  1  dose[[d]]:  30 
i:  16  tmp_dose:  30  d:  1  dose[[d]]:  30 
specific_gravity_adjustment : units not recognized!i:  17  tmp_dose:  200  d:  1  dose[[d]]:  200 
i:  17  tmp_dose:  200  d:  1  dose[[d]]:  200 
specific_gravity_adjustment : units not recognized!i:  18  tmp_dose:  100  d:  1  dose[[d]]:  100 
i:  18  tmp_dose:  100  d:  1  dose[[d]]:  100 
specific_gravity_adjustment : units not recognized!i:  19  tmp_dose:  200  d:  1  dose[[d]]:  200 
i:  19  tmp_dose:  200  d:  1  dose[[d]]:  200 
specific_gravity_adjustment : units not recognized!i:  20  tmp_dose:  200  d:  1  dose[[d]]:  200 
i:  20  tmp_dose:  200  d:  1  dose[[d]]:  200 
specific_gravity_adjustment : units not recognized!i:  21  tmp_dose:  100  d:  1  dose[[d]]:  100 
i:  21  tmp_dose:  100  d:  1  dose[[d]]:  100 
specific_gravity_adjustment : units not recognized!i:  22  tmp_dose:  200  d:  1  dose[[d]]:  200 
i:  22  tmp_dose:  200  d:  1  dose[[d]]:  200 
specific_gravity_adjustment : units not recognized!i:  23  tmp_dose:  100  d:  1  dose[[d]]:  100 
i:  23  tmp_dose:  100  d:  1  dose[[d]]:  100 
specific_gravity_adjustment : units not recognized!i:  24  tmp_dose:  100  d:  1  dose[[d]]:  100 
i:  24  tmp_dose:  100  d:  1  dose[[d]]:  100 
specific_gravity_adjustment : units not recognized!i:  25  tmp_dose:  200  d:  1  dose[[d]]:  200 
i:  25  tmp_dose:  200  d:  1  dose[[d]]:  200 
specific_gravity_adjustment : units not recognized!i:  26  tmp_dose:  100  d:  1  dose[[d]]:  100 
i:  26  tmp_dose:  100  d:  1  dose[[d]]:  100 
specific_gravity_adjustment : units not recognized!i:  27  tmp_dose:  100  d:  1  dose[[d]]:  100 
i:  27  tmp_dose:  100  d:  1  dose[[d]]:  100 
specific_gravity_adjustment : units not recognized!i:  28  tmp_dose:  100  d:  1  dose[[d]]:  100 
i:  28  tmp_dose:  100  d:  1  dose[[d]]:  100 
specific_gravity_adjustment : units not recognized!i:  29  tmp_dose:  200  d:  1  dose[[d]]:  200 
i:  29  tmp_dose:  200  d:  1  dose[[d]]:  200 
specific_gravity_adjustment : units not recognized!i:  30  tmp_dose:  30  d:  1  dose[[d]]:  30 
i:  30  tmp_dose:  30  d:  1  dose[[d]]:  30 
specific_gravity_adjustment : units not recognized!i:  31  tmp_dose:  200  d:  1  dose[[d]]:  200 
i:  31  tmp_dose:  200  d:  1  dose[[d]]:  200 
specific_gravity_adjustment : units not recognized!i:  32  tmp_dose:  30  d:  1  dose[[d]]:  30 
i:  32  tmp_dose:  30  d:  1  dose[[d]]:  30 
specific_gravity_adjustment : units not recognized!i:  33  tmp_dose:  200  d:  1  dose[[d]]:  200 
i:  33  tmp_dose:  200  d:  1  dose[[d]]:  200 
specific_gravity_adjustment : units not recognized!i:  34  tmp_dose:  30  d:  1  dose[[d]]:  30 
i:  34  tmp_dose:  30  d:  1  dose[[d]]:  30 
specific_gravity_adjustment : units not recognized!i:  35  tmp_dose:  100  d:  1  dose[[d]]:  100 
i:  35  tmp_dose:  100  d:  1  dose[[d]]:  100 
specific_gravity_adjustment : units not recognized!i:  36  tmp_dose:  200  d:  1  dose[[d]]:  200 
i:  36  tmp_dose:  200  d:  1  dose[[d]]:  200 
specific_gravity_adjustment : units not recognized!i:  37  tmp_dose:  100  d:  1  dose[[d]]:  100 
i:  37  tmp_dose:  100  d:  1  dose[[d]]:  100 
specific_gravity_adjustment : units not recognized!i:  38  tmp_dose:  100  d:  1  dose[[d]]:  100 
i:  38  tmp_dose:  100  d:  1  dose[[d]]:  100 
specific_gravity_adjustment : units not recognized!i:  39  tmp_dose:  100  d:  1  dose[[d]]:  100 
i:  39  tmp_dose:  100  d:  1  dose[[d]]:  100 
specific_gravity_adjustment : units not recognized!i:  40  tmp_dose:  100  d:  1  dose[[d]]:  100 
i:  40  tmp_dose:  100  d:  1  dose[[d]]:  100 
specific_gravity_adjustment : units not recognized!i:  41  tmp_dose:  200  d:  1  dose[[d]]:  200 
i:  41  tmp_dose:  200  d:  1  dose[[d]]:  200 
specific_gravity_adjustment : units not recognized!i:  42  tmp_dose:  30  d:  1  dose[[d]]:  30 
i:  42  tmp_dose:  30  d:  1  dose[[d]]:  30 
specific_gravity_adjustment : units not recognized!i:  43  tmp_dose:  100  d:  1  dose[[d]]:  100 
i:  43  tmp_dose:  100  d:  1  dose[[d]]:  100 
specific_gravity_adjustment : units not recognized!i:  44  tmp_dose:  100  d:  1  dose[[d]]:  100 
i:  44  tmp_dose:  100  d:  1  dose[[d]]:  100 
specific_gravity_adjustment : units not recognized!i:  45  tmp_dose:  100  d:  1  dose[[d]]:  100 
i:  45  tmp_dose:  100  d:  1  dose[[d]]:  100 
unit_conversion : Unit Class 1 (Time) time_col:  TLAG KELTMLO KELTMHI THALF MIDPT1 MIDPT2 TAU1 TOLD1  parameters are scaled from  HR  to  HR  via scaling factor:  1 
unit_conversion : Unit Class 2 (Amount) amount_col:  AMT.12.00 AMT.24.00 AE.12.00 AE.24.00 AE  parameters are scaled from  NG  to  MG  via scaling factor:  1e-06 
unit_conversion : Unit Class 3 (Dose) dose_col:  DOSE1  parameters are scaled from  MG  to  MG  via scaling factor:  1 
unit_conversion : Unit Class 4 (Volume) volume_col:  VOLSUM  parameters are scaled from  ML  to  L  via scaling factor:  0.001 
unit_conversion : Unit Class 6 (1/Time) kel_col:  KEL  parameters are scaled from  1/HR  to  1/HR  via scaling factor:  1 
unit_conversion : Unit Class 11: ([Volume.Amount]/Volume) aurc_col:  AURCALL AURCLAST AURCINFO AURCINFP  parameters are scaled from  ML.NG/ML  to  ML.UG/ML  via scaling factor:  0.001 
unit_conversion : Unit Class 13: (Amount/Time) rate_col:  MAXRATE1 RATELAST1 RATE1 RATE2  parameters are scaled from  NG/HR  to  UG/HR  via scaling factor:  0.001 
There were 50 or more warnings (use warnings() to see the first 50)

Warning messages:
1: In run_M4_SS_computation(data = merged_data, map = map_data,  ... :
  Flag 'FLGEMESIS' is not present in the dataset
2: In specific_gravity_adjustment(amt_units = volu, map = map) :
  NAs introduced by coercion
3: In specific_gravity_adjustment(amt_units = volu, map = map) : 
4: In specific_gravity_adjustment(amt_units = volu, map = map) :
  NAs introduced by coercion
5: In specific_gravity_adjustment(amt_units = volu, map = map) : 
6: In specific_gravity_adjustment(amt_units = volu, map = map) :
  NAs introduced by coercion
7: In specific_gravity_adjustment(amt_units = volu, map = map) : 
8: In specific_gravity_adjustment(amt_units = volu, map = map) :
  NAs introduced by coercion
9: In specific_gravity_adjustment(amt_units = volu, map = map) : 
10: In specific_gravity_adjustment(amt_units = volu, map = map) :
  NAs introduced by coercion
11: In specific_gravity_adjustment(amt_units = volu, map = map) : 
12: In specific_gravity_adjustment(amt_units = volu, map = map) :
  NAs introduced by coercion
13: In specific_gravity_adjustment(amt_units = volu, map = map) : 
14: In specific_gravity_adjustment(amt_units = volu, map = map) :
  NAs introduced by coercion
15: In specific_gravity_adjustment(amt_units = volu, map = map) : 
16: In specific_gravity_adjustment(amt_units = volu, map = map) :
  NAs introduced by coercion
17: In specific_gravity_adjustment(amt_units = volu, map = map) : 
18: In specific_gravity_adjustment(amt_units = volu, map = map) :
  NAs introduced by coercion
19: In specific_gravity_adjustment(amt_units = volu, map = map) : 
20: In specific_gravity_adjustment(amt_units = volu, map = map) :
  NAs introduced by coercion
21: In specific_gravity_adjustment(amt_units = volu, map = map) : 
22: In specific_gravity_adjustment(amt_units = volu, map = map) :
  NAs introduced by coercion
23: In specific_gravity_adjustment(amt_units = volu, map = map) : 
24: In specific_gravity_adjustment(amt_units = volu, map = map) :
  NAs introduced by coercion
25: In specific_gravity_adjustment(amt_units = volu, map = map) : 
26: In specific_gravity_adjustment(amt_units = volu, map = map) :
  NAs introduced by coercion
27: In specific_gravity_adjustment(amt_units = volu, map = map) : 
28: In specific_gravity_adjustment(amt_units = volu, map = map) :
  NAs introduced by coercion
29: In specific_gravity_adjustment(amt_units = volu, map = map) : 
30: In specific_gravity_adjustment(amt_units = volu, map = map) :
  NAs introduced by coercion
31: In specific_gravity_adjustment(amt_units = volu, map = map) : 
32: In specific_gravity_adjustment(amt_units = volu, map = map) :
  NAs introduced by coercion
33: In specific_gravity_adjustment(amt_units = volu, map = map) : 
34: In specific_gravity_adjustment(amt_units = volu, map = map) :
  NAs introduced by coercion
35: In specific_gravity_adjustment(amt_units = volu, map = map) : 
36: In specific_gravity_adjustment(amt_units = volu, map = map) :
  NAs introduced by coercion
37: In specific_gravity_adjustment(amt_units = volu, map = map) : 
38: In specific_gravity_adjustment(amt_units = volu, map = map) :
  NAs introduced by coercion
39: In specific_gravity_adjustment(amt_units = volu, map = map) : 
40: In specific_gravity_adjustment(amt_units = volu, map = map) :
  NAs introduced by coercion
41: In specific_gravity_adjustment(amt_units = volu, map = map) : 
42: In specific_gravity_adjustment(amt_units = volu, map = map) :
  NAs introduced by coercion
43: In specific_gravity_adjustment(amt_units = volu, map = map) : 
44: In specific_gravity_adjustment(amt_units = volu, map = map) :
  NAs introduced by coercion
45: In specific_gravity_adjustment(amt_units = volu, map = map) : 
46: In specific_gravity_adjustment(amt_units = volu, map = map) :
  NAs introduced by coercion
47: In specific_gravity_adjustment(amt_units = volu, map = map) : 
48: In specific_gravity_adjustment(amt_units = volu, map = map) :
  NAs introduced by coercion
49: In specific_gravity_adjustment(amt_units = volu, map = map) : 
50: In specific_gravity_adjustment(amt_units = volu, map = map) :
  NAs introduced by coercion

```

**Comment by rudraya** [comment]
This issue was addressed in 96f6417, please let me know if it still occurs.

**Comment by tensfeldt** [comment]
This particular issue #33 has been addressed with tc018 in commit 96f6417 and can be closed
